### PR TITLE
librbd/cmake: Add missed PWL related types

### PR DESCRIPTION
**Pull request by lixiaoy1**
Add PWL related classes into rbd_types. And fix the ceph-dencoder build issue when both SSD and RWL are enabled.

Signed-off-by: Li, Xiaoyan <xiaoyan.li@intel.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/librbd/CMakeLists.txt [hunk 1]
```diff
@@ -9,9 +9,18 @@ set(librbd_types_srcs
...
   WatchNotifyTypes.cc)
 
+if(WITH_RBD_RWL OR WITH_RBD_SSD_CACHE)
+ list(APPEND librbd_types_srcs cache/pwl/Types.cc)
+endif()
+
 add_library(rbd_types STATIC
   ${librbd_types_srcs})
 
+if (WITH_RBD_RWL AND WITH_RBD_SSD_CACHE)
```
Comment (dillaman) [inline src/librbd/CMakeLists.txt:19]: Nit: no need for `pmem` if just SSD, right?

Comment (lixiaoy1) [reply src/librbd/CMakeLists.txt:19]: Yes, as no encoding/decoding for pmem.

Comment (dillaman) [reply src/librbd/CMakeLists.txt:19]: ... so should it be `if (WITH_RBD_SSD_CACHE)`?

Comment (dillaman) [reply src/librbd/CMakeLists.txt:19]: Ack -- I understand now. Thx.

**Review by dillaman — APPROVED** [review]

**Pull request merged**